deeplinkdispatch-base/src/main/java/com/airbnb/deeplinkdispatch/UrlTree.kt

```diff
@@ -128,6 +140,50 @@ data class Host(override val id: String) : TreeNode(id = id, metadata = NodeMeta
 
 data class PathSegment(override val id: String) : TreeNode(id = id, metadata = NodeMetadata(MetadataMasks.ComponentTypePathSegmentMask, id))
 
+/**
+ * Match data byte array format is:
+ * 0..1 url                                                        url template length
+ * 2..(2+url template length)                                      url template
+ * (2+url template length)..
+ * (2+url template length)+2                                       classname length
+ * (2+url template length)+2..
+ * (2+url template length)+2+classname length                      class name
+ * (2+url template length)+2+classname length..
+ * (2+url template length)+2+classname length+1                    method name length (can be 0)
+ * (2+url template length)+2+classname length+1..
+ * (2+url template length)+2+classname length+1+method name length method name
+ */
+fun matchByteArray(match: UriMatch?): UByteArray {
+    return match?.let {
```

Comment:
avoid the nesting and using `it` in such  large block
```suggestion
    if (match == null) return UByteArray(0)
```